app/utils/theme.ts

```diff
@@ -2,20 +2,21 @@ import { Appearance } from 'react-native-appearance';
 import changeNavigationBarColor from 'react-native-navigation-bar-color';
 import setRootViewColor from 'rn-root-view';
 
+import { IThemePreference } from '../definitions/ITheme';
 import { themes } from '../constants/colors';
 import { isAndroid } from './deviceInfo';
 
-let themeListener;
+let themeListener: { remove: () => void } | null;
 
-export const defaultTheme = () => {
+export const defaultTheme = (): string => {
```

Comment:
I think we can create an type for theme at `definitions/ITheme.ts` for example:
```
export type TThemeMode = 'automatic' | 'light' | 'dark';

export interface IThemePreference {
	currentTheme: TThemeMode;
	darkLevel: string;
}
```

and with that, we can change those interfaces:
* `app/share.tsx` line 40: `themePreferences: IThemePreference;` 
* `app/index.tsx` line 46: `themePreferences: IThemePreference;` 
* `app/theme.tsx` line 8: `themePreferences: IThemePreference;`